### ISSUE TITLE
Error: New version 1.16.1 is not signed by the application owner

### ISSUE THREAD

**Issue by ghost** [buttercup/buttercup-desktop#788]
Hi, I tried to update from 1.16.0 to 1.16.1 but got this error:

Unhandled Promise Rejection
Error: New version 1.16.1 is not signed by the application owner: publisherNames: COMODO RSA Code Signing CA, raw info: {
  "SignerCertificate": {
    "FriendlyName": "",
    "IssuerName": {
      "Name": "CN=COMODO RSA Code Signing CA, O=COMODO CA Limited, L=Salford, S=Greater Manchester, C=GB",
      "Oid": "System.Security.Cryptography.Oid"
    },
    "NotAfter": "/Date(1578009599000)/",
    "NotBefore": "/Date(1546387200000)/",
    "PrivateKey": null,
    "PublicKey": {
      "Key": "System.Security.Cryptography.RSACryptoServiceProvider",
      "Oid": "System.Security.Cryptography.Oid",
      "EncodedKeyValue": "System.Security.Cryptography.AsnEncodedData",
      "EncodedParameters": "System.Security.Cryptography.AsnEncodedData"
    },
    "SerialNumber": "00BDC2D8815A5B41D6045C6C719D8A8F2A",
    "SignatureAlgorithm": {
      "Value": "1.2.840.113549.1.1.11",
      "FriendlyName": "sha256RSA"
    },
    "Thumbprint": "A9C56AD137B3A5359A375A1FA4827DDD071A845A",
    "Version": 3,
    "Issuer": "CN=COMODO RSA Code Signing CA, O=COMODO CA Limited, L=Salford, S=Greater Manchester, C=GB",
    "Subject": "CN=MadDev Oy, O=MadDev Oy, STREET=N�kinkaari 4 A, L=Espoo, S=Uusimaa, PostalCode=02320, C=FI"
  },
  "TimeStamperCertificate": {
    "Archived": false,
    "Extensions": [
      "System.Security.Cryptography.X509Certificates.X509KeyUsageExtension",
      "System.Security.Cryptography.X509Certificates.X509BasicConstraintsExtension",
      "System.Security.Cryptography.X509Certificates.X509EnhancedKeyUsageExtension",
      "System.Security.Cryptography.X509Certificates.X509Extension",
      "System.Security.Cryptography.X509Certificates.X509Extension",
      "System.Security.Cryptography.X509Certificates.X509SubjectKeyIdentifierExtension",
      "System.Security.Cryptography.X509Certificates.X509Extension",
      "System.Security.Cryptography.X509Certificates.X509Extension"
    ],
    "FriendlyName": "",
    "IssuerName": {
      "Name": "CN=DigiCert Assured ID CA-1, OU=www.digicert.com, O=DigiCert Inc, C=US",
      "Oid": "System.Security.Cryptography.Oid"
    },
    "NotAfter": "/Date(1729555200000)/",
    "NotBefore": "/Date(1413936000000)/",
    "HasPrivateKey": false,
    "PrivateKey": null,
    "PublicKey": {
      "Key": "System.Security.Cryptography.RSACryptoServiceProvider",
      "Oid": "System.Security.Cryptography.Oid",
      "EncodedKeyValue": "System.Security.Cryptography.AsnEncodedData",
      "EncodedParameters": "System.Security.Cryptography.AsnEncodedData"
    },
    "SerialNumber": "03019A023AFF58B16BD6D5EAE617F066",
    "SubjectName": {
      "Name": "CN=DigiCert Timestamp Responder, O=DigiCert, C=US",
      "Oid": "System.Security.Cryptography.Oid"
    },
    "SignatureAlgorithm": {
      "Value": "1.2.840.113549.1.1.5",
      "FriendlyName": "sha1RSA"
    },
    "Thumbprint": "614D271D9102E30169822487FDE5DE00A352B01D",
    "Version": 3,
    "Handle": 1923585426592,
    "Issuer": "CN=DigiCert Assured ID CA-1, OU=www.digicert.com, O=DigiCert Inc, C=US",
    "Subject": "CN=DigiCert Timestamp Responder, O=DigiCert, C=US"
  },
  "Status": 0,
  "StatusMessage": "Signature verified."
}
    at t.newError (C:/Program Files/Buttercup/resources/app.asar/dist/app.js:1:8496)
    at Object.<anonymous> (C:/Program Files/Buttercup/resources/app.asar/dist/app.js:207:228505)
    at Generator.next (<anonymous>)
    at Generator.c (C:/Program Files/Buttercup/resources/app.asar/dist/app.js:1:1411)
    at d._promiseFulfilled (C:/Program Files/Buttercup/resources/app.asar/dist/app.js:206:189112)
    at j._settlePromise (C:/Program Files/Buttercup/resources/app.asar/dist/app.js:206:145545)
    at j._settlePromise0 (C:/Program Files/Buttercup/resources/app.asar/dist/app.js:206:146143)
    at j._settlePromises (C:/Program Files/Buttercup/resources/app.asar/dist/app.js:206:147495)
    at d (C:/Program Files/Buttercup/resources/app.asar/dist/app.js:206:149218)
    at p (C:/Program Files/Buttercup/resources/app.asar/dist/app.js:206:149157)
    at s._drainQueues (C:/Program Files/Buttercup/resources/app.asar/dist/app.js:206:150677)
    at Immediate.drainQueues [as _onImmediate] (C:/Program Files/Buttercup/resources/app.asar/dist/app.js:206:148899)
From previous event:
    at j.z [as _captureStackTrace] (C:/Program Files/Buttercup/resources/app.asar/dist/app.js:206:164744)
    at new d (C:/Program Files/Buttercup/resources/app.asar/dist/app.js:206:187987)
    at C:/Program Files/Buttercup/resources/app.asar/dist/app.js:206:191076
    at t.NsisUpdater.downloadUpdate (C:/Program Files/Buttercup/resources/app.asar/dist/app.js:71:20764)
    at EventEmitter.i.ipcMain.on (C:/Program Files/Buttercup/resources/app.asar/dist/app.js:414:129520)
    at WebContents.<anonymous> (C:/Program Files/Buttercup/resources/electron.asar/browser/api/web-contents.js:285:13)

**Comment by jy95** [comment]
Me too - It is strange : It was fixed in 1.16.0 .... https://github.com/buttercup/buttercup-desktop/issues/779

**Comment by perry-mitchell** [comment]
Strange, I’ve not seen this issue myself nor have others using it.. 😕

**Comment by ottopotto** [comment]
Don't know if this related and if this is normal, but before the 'not signed' error differential download fails:

```javascript
[2019-07-18 16:08:30.349] [info] Downloading update from buttercup-desktop-setup-1.16.1.exe
[2019-07-18 16:08:30.361] [info] No cached update info available
[2019-07-18 16:08:30.364] [info] Download block maps (old: "https://github.com/buttercup/buttercup-desktop/releases/download/v1.16.0/buttercup-desktop-setup-1.16.1.exe.blockmap", new: https://github.com/buttercup/buttercup-desktop/releases/download/v1.16.1/buttercup-desktop-setup-1.16.1.exe.blockmap)
[2019-07-18 16:08:31.266] [error] Cannot download differentially, fallback to full download: HttpError: 404 Not Found
"method: undefined url: https://github.com/buttercup/buttercup-desktop/releases/download/v1.16.0/buttercup-desktop-setup-1.16.1.exe.blockmap\n\nPlease double check that your authentication token is correct. Due to security reasons actual status maybe not reported, but 404.\n"
Headers: {
  "cache-control": [
    "no-cache"
  ],
  "content-encoding": [
    "gzip"
  ],
  "content-security-policy": [
    "default-src 'none'; base-uri 'self'; connect-src 'self'; form-action 'self'; img-src 'self' data:; script-src 'self'; style-src 'unsafe-inline'"
  ],
  "content-type": [
    "text/plain; charset=utf-8"
  ],
  "date": [
    "Thu, 18 Jul 2019 13:08:32 GMT"
  ],
  "expect-ct": [
    "max-age=2592000, report-uri=\"https://api.github.com/_private/browser/errors\""
  ],
  "server": [
    "GitHub.com"
  ],
  "set-cookie": [
    "has_recent_activity=1; path=/; expires=Thu, 18 Jul 2019 14:08:32 -0000",
    "logged_in=no; domain=.github.com; path=/; expires=Mon, 18 Jul 2039 13:08:32 -0000; secure; HttpOnly",
    "_gh_sess=UGZhNE1seEZUNEdRcEtNdXVzV2FmQmNqZG5vSlpmRGhBM3VFTk1WTitxc055MlhmR0NOUWZBZ2RjTk5XR2MzNGFTY1NHWEhhV1FtbTMrTUhLRzR4cWRTejgwZWp6RHhKKzFsUkUwTkRUeGJoSnowWHo0a2FmOEtPeXQ2MktjemovTFZxSWlpWlUyUkRFTTNpMmN5Wi9tU3pyL0VDVFFkUUpBNE80TStGSndkMytaQ3M5cjVCZHV5aVhGUEZGNTBQQzk3MjUwcG42VHQ2VjVBOENIWjdpZz09LS0vc3orQkVidmlFalhkQlZVLzdpV2RnPT0%3D--a64b755813781ba40c6a79d15a26ee0275cd14d3; path=/; secure; HttpOnly"
  ],
  "status": [
    "404 Not Found"
  ],
  "strict-transport-security": [
    "max-age=31536000; includeSubdomains; preload"
  ],
  "transfer-encoding": [
    "chunked"
  ],
  "vary": [
    "X-PJAX"
  ],
  "x-content-type-options": [
    "nosniff"
  ],
  "x-frame-options": [
    "deny"
  ],
  "x-github-request-id": [
    "6779:4395C:A56BE1:100F253:5D306F4F"
  ],
  "x-request-id": [
    "e72ebedc-6b36-4bc1-9997-65b1e75df63f"
  ],
  "x-xss-protection": [
    "1; mode=block"
  ]
}
    at f (C:\Program Files\Buttercup\resources\app.asar\dist\app.js:96:7070)
    at t.ElectronHttpExecutor.handleResponse (C:\Program Files\Buttercup\resources\app.asar\dist\app.js:96:8665)
    at ClientRequest.s.doRequest.o (C:\Program Files\Buttercup\resources\app.asar\dist\app.js:96:8134)
    at ClientRequest.emit (events.js:182:13)
    at URLRequest.ClientRequest.urlRequest.on (C:\Program Files\Buttercup\resources\electron.asar\browser\api\net.js:207:12)
    at URLRequest.emit (events.js:182:13)
From previous event:
    at j.z [as _captureStackTrace] (C:\Program Files\Buttercup\resources\app.asar\dist\app.js:206:164744)
    at new d (C:\Program Files\Buttercup\resources\app.asar\dist\app.js:206:187987)
    at C:\Program Files\Buttercup\resources\app.asar\dist\app.js:206:191076
    at t.NsisUpdater.downloadUpdate (C:\Program Files\Buttercup\resources\app.asar\dist\app.js:71:20764)
    at EventEmitter.i.ipcMain.on (C:\Program Files\Buttercup\resources\app.asar\dist\app.js:414:129520)
    at EventEmitter.emit (events.js:182:13)
    at WebContents.<anonymous> (C:\Program Files\Buttercup\resources\electron.asar\browser\api\web-contents.js:285:13)
    at WebContents.emit (events.js:182:13)
[2019-07-18 16:08:53.642] [warn] Sign verification failed, installer signed with incorrect certificate: publisherNames: COMODO RSA Code Signing CA, raw info: {
  "SignerCertificate": {
    "FriendlyName": "",
    "IssuerName": {
      "Name": "CN=COMODO RSA Code Signing CA, O=COMODO CA Limited, L=Salford, S=Greater Manchester, C=GB",
      "Oid": "System.Security.Cryptography.Oid"
    },
    "NotAfter": "/Date(1578009599000)/",
    "NotBefore": "/Date(1546387200000)/",
    "PrivateKey": null,
    "PublicKey": {
      "Key": "System.Security.Cryptography.RSACryptoServiceProvider",
      "Oid": "System.Security.Cryptography.Oid",
      "EncodedKeyValue": "System.Security.Cryptography.AsnEncodedData",
      "EncodedParameters": "System.Security.Cryptography.AsnEncodedData"
    },
    "SerialNumber": "00BDC2D8815A5B41D6045C6C719D8A8F2A",
    "SignatureAlgorithm": {
      "Value": "1.2.840.113549.1.1.11",
      "FriendlyName": "sha256RSA"
    },
    "Thumbprint": "A9C56AD137B3A5359A375A1FA4827DDD071A845A",
    "Version": 3,
    "Issuer": "CN=COMODO RSA Code Signing CA, O=COMODO CA Limited, L=Salford, S=Greater Manchester, C=GB",
    "Subject": "CN=MadDev Oy, O=MadDev Oy, STREET=N�kinkaari 4 A, L=Espoo, S=Uusimaa, PostalCode=02320, C=FI"
  },
  "TimeStamperCertificate": {
    "Archived": false,
    "Extensions": [
      "System.Security.Cryptography.X509Certificates.X509KeyUsageExtension",
      "System.Security.Cryptography.X509Certificates.X509BasicConstraintsExtension",
      "System.Security.Cryptography.X509Certificates.X509EnhancedKeyUsageExtension",
      "System.Security.Cryptography.X509Certificates.X509Extension",
      "System.Security.Cryptography.X509Certificates.X509Extension",
      "System.Security.Cryptography.X509Certificates.X509SubjectKeyIdentifierExtension",
      "System.Security.Cryptography.X509Certificates.X509Extension",
      "System.Security.Cryptography.X509Certificates.X509Extension"
    ],
    "FriendlyName": "",
    "IssuerName": {
      "Name": "CN=DigiCert Assured ID CA-1, OU=www.digicert.com, O=DigiCert Inc, C=US",
      "Oid": "System.Security.Cryptography.Oid"
    },
    "NotAfter": "/Date(1729555200000)/",
    "NotBefore": "/Date(1413936000000)/",
    "HasPrivateKey": false,
    "PrivateKey": null,
    "PublicKey": {
      "Key": "System.Security.Cryptography.RSACryptoServiceProvider",
      "Oid": "System.Security.Cryptography.Oid",
      "EncodedKeyValue": "System.Security.Cryptography.AsnEncodedData",
      "EncodedParameters": "System.Security.Cryptography.AsnEncodedData"
    },
    "SerialNumber": "03019A023AFF58B16BD6D5EAE617F066",
    "SubjectName": {
      "Name": "CN=DigiCert Timestamp Responder, O=DigiCert, C=US",
      "Oid": "System.Security.Cryptography.Oid"
    },
    "SignatureAlgorithm": {
      "Value": "1.2.840.113549.1.1.5",
      "FriendlyName": "sha1RSA"
    },
    "Thumbprint": "614D271D9102E30169822487FDE5DE00A352B01D",
    "Version": 3,
    "Handle": 1593085405872,
    "Issuer": "CN=DigiCert Assured ID CA-1, OU=www.digicert.com, O=DigiCert Inc, C=US",
    "Subject": "CN=DigiCert Timestamp Responder, O=DigiCert, C=US"
  },
  "Status": 0,
  "StatusMessage": "Signature verified."
}
[2019-07-18 16:08:53.662] [error] Error: Error: New version 1.16.1 is not signed by the application owner: publisherNames: COMODO RSA Code Signing CA, raw info: {
```

**Comment by sallar** [comment]
Hi @jy95 and @ottopotto could you please test this build?
https://www.icloud.com/iclouddrive/0d4r6jshKXg-zV5N0EoiwV_KA#Buttercup_Setup_1.16.2
I have resigned the 1.16.1 without change but updated the version.

Please let me know if this worked, in that case I can release this publicly.

**Comment by ottopotto** [comment]
Works for me.

I never tried installing 1.16.1 manually though. I only used the Update button in GUI when prompted. Maybe installing 1.16.1 manually would have worked as well?

**Comment by jy95** [comment]
I manually installed the 1.16.1 from the github and it works as expected.
Obviously, something is wrong in the upgrader ...

**Comment by ottopotto** [comment]
Maybe worth a mention, I also had trouble updating from 1.50 to 1.60 earlier. I retried it now (install 1.15.0 and update to 1.16.1) and this is the error in that case:
![errorspawn](https://user-images.githubusercontent.com/1174957/61489992-9aede480-a9b4-11e9-9e0c-ec2feeb4cd48.PNG)

There are no error logs:
```
[2019-07-18 23:15:52:0365] [info] Checking for update
[2019-07-18 23:15:54:0177] [info] Found version 1.16.1 (url: buttercup-desktop-setup-1.16.1.exe)
[2019-07-18 23:15:56:0956] [info] Downloading update from buttercup-desktop-setup-1.16.1.exe
[2019-07-18 23:16:24:0456] [info] New version 1.16.1 has been downloaded to C:\Users\Otto\AppData\Local\Temp\up-gcQgCO\buttercup-desktop-setup-1.16.1.exe
[2019-07-18 23:16:24:0456] [info] Install on explicit quitAndInstall
[2019-07-18 23:16:24:0456] [info] Install: isSilent: false, isRunAfter: true
[2019-07-18 23:16:29:0540] [info] Running before-quit operation.
```

If you run that exe in AppData\Local\Temp manually from File Explorer, it works ok.

On the other hand, when trying to update 1.16.0->1.16.1, only an ICO file is created in that Temp directory and the 'not signed' error comes up.

BTW regarding the differential download error I mentioned, is the path in the first URL incorrect, `https://github.com/buttercup/buttercup-desktop/releases/download/v1.16.0/buttercup-desktop-setup-1.16.1.exe.blockmap` - directory v1.16.0 instead of v1.16.1? Can it be related to the 'not signed' error?

**Comment by sallar** [comment]
@ottopotto @jy95 version 1.16.2 is released now, would you be able to upgrade to see if it works now?

**Comment by jy95** [comment]
Testing that right now : It looks like the upgrade processus is freezing : 

![image](https://user-images.githubusercontent.com/9306961/61592956-d74e5a00-abd9-11e9-97ee-6cd65d678287.png)

**Comment by ottopotto** [comment]
It doesn't work for me either. Same result as before.

**Comment by Shahzaky** [comment]
Got the same issue. Tried to update from 1.16.1 to 1.16.2:

```
Unhandled Promise Rejection

Error: New version 1.16.2 is not signed by the application owner: publisherNames: COMODO RSA Code Signing CA, raw info: {
  "SignerCertificate": {
    "FriendlyName": "",
    "IssuerName": {
      "Name": "CN=COMODO RSA Code Signing CA, O=COMODO CA Limited, L=Salford, S=Greater Manchester, C=GB",
      "Oid": "System.Security.Cryptography.Oid"
    },
    "NotAfter": "/Date(1578009599000)/",
    "NotBefore": "/Date(1546387200000)/",
    "PrivateKey": null,
    "PublicKey": {
      "Key": "System.Security.Cryptography.RSACryptoServiceProvider",
      "Oid": "System.Security.Cryptography.Oid",
      "EncodedKeyValue": "System.Security.Cryptography.AsnEncodedData",
      "EncodedParameters": "System.Security.Cryptography.AsnEncodedData"
    },
    "SerialNumber": "00BDC2D8815A5B41D6045C6C719D8A8F2A",
    "SignatureAlgorithm": {
      "Value": "1.2.840.113549.1.1.11",
      "FriendlyName": "sha256RSA"
    },
    "Thumbprint": "A9C56AD137B3A5359A375A1FA4827DDD071A845A",
    "Version": 3,
    "Issuer": "CN=COMODO RSA Code Signing CA, O=COMODO CA Limited, L=Salford, S=Greater Manchester, C=GB",
    "Subject": "CN=MadDev Oy, O=MadDev Oy, STREET=N�kinkaari 4 A, L=Espoo, S=Uusimaa, PostalCode=02320, C=FI"
  },
  "TimeStamperCertificate": {
    "Archived": false,
    "Extensions": [
      "System.Security.Cryptography.X509Certificates.X509KeyUsageExtension",
      "System.Security.Cryptography.X509Certificates.X509BasicConstraintsExtension",
      "System.Security.Cryptography.X509Certificates.X509EnhancedKeyUsageExtension",
      "System.Security.Cryptography.X509Certificates.X509Extension",
      "System.Security.Cryptography.X509Certificates.X509Extension",
      "System.Security.Cryptography.X509Certificates.X509SubjectKeyIdentifierExtension",
      "System.Security.Cryptography.X509Certificates.X509Extension",
      "System.Security.Cryptography.X509Certificates.X509Extension"
    ],
    "FriendlyName": "",
    "IssuerName": {
      "Name": "CN=DigiCert Assured ID CA-1, OU=www.digicert.com, O=DigiCert Inc, C=US",
      "Oid": "System.Security.Cryptography.Oid"
    },
    "NotAfter": "/Date(1729555200000)/",
    "NotBefore": "/Date(1413936000000)/",
    "HasPrivateKey": false,
    "PrivateKey": null,
    "PublicKey": {
      "Key": "System.Security.Cryptography.RSACryptoServiceProvider",
      "Oid": "System.Security.Cryptography.Oid",
      "EncodedKeyValue": "System.Security.Cryptography.AsnEncodedData",
      "EncodedParameters": "System.Security.Cryptography.AsnEncodedData"
    },
    "SerialNumber": "03019A023AFF58B16BD6D5EAE617F066",
    "SubjectName": {
      "Name": "CN=DigiCert Timestamp Responder, O=DigiCert, C=US",
      "Oid": "System.Security.Cryptography.Oid"
    },
    "SignatureAlgorithm": {
      "Value": "1.2.840.113549.1.1.5",
      "FriendlyName": "sha1RSA"
    },
    "Thumbprint": "614D271D9102E30169822487FDE5DE00A352B01D",
    "Version": 3,
    "Handle": 1865019313120,
    "Issuer": "CN=DigiCert Assured ID CA-1, OU=www.digicert.com, O=DigiCert Inc, C=US",
    "Subject": "CN=DigiCert Timestamp Responder, O=DigiCert, C=US"
  },
  "Status": 0,
  "StatusMessage": "Signature verified."
}
    at t.newError (C:/Program Files/Buttercup/resources/app.asar/dist/app.js:1:8496)
    at Object.<anonymous> (C:/Program Files/Buttercup/resources/app.asar/dist/app.js:207:228505)
    at Generator.next (<anonymous>)
    at Generator.c (C:/Program Files/Buttercup/resources/app.asar/dist/app.js:1:1411)
    at d._promiseFulfilled (C:/Program Files/Buttercup/resources/app.asar/dist/app.js:206:189112)
    at j._settlePromise (C:/Program Files/Buttercup/resources/app.asar/dist/app.js:206:145545)
    at j._settlePromise0 (C:/Program Files/Buttercup/resources/app.asar/dist/app.js:206:146143)
    at j._settlePromises (C:/Program Files/Buttercup/resources/app.asar/dist/app.js:206:147495)
    at d (C:/Program Files/Buttercup/resources/app.asar/dist/app.js:206:149218)
    at p (C:/Program Files/Buttercup/resources/app.asar/dist/app.js:206:149157)
    at s._drainQueues (C:/Program Files/Buttercup/resources/app.asar/dist/app.js:206:150677)
    at Immediate.drainQueues [as _onImmediate] (C:/Program Files/Buttercup/resources/app.asar/dist/app.js:206:148899)
From previous event:
    at j.z [as _captureStackTrace] (C:/Program Files/Buttercup/resources/app.asar/dist/app.js:206:164744)
    at new d (C:/Program Files/Buttercup/resources/app.asar/dist/app.js:206:187987)
    at C:/Program Files/Buttercup/resources/app.asar/dist/app.js:206:191076
    at t.NsisUpdater.downloadUpdate (C:/Program Files/Buttercup/resources/app.asar/dist/app.js:71:20764)
    at EventEmitter.i.ipcMain.on (C:/Program Files/Buttercup/resources/app.asar/dist/app.js:414:129520)
    at WebContents.<anonymous> (C:/Program Files/Buttercup/resources/electron.asar/browser/api/web-contents.js:285:13)
```

**Comment by perry-mitchell** [comment]
Code signing is implemented, but expired. Being fixed in #1003
